packages/clerk-js/src/core/resources/SignIn.ts

```diff
@@ -244,12 +248,27 @@ export class SignIn extends BaseResource implements SignInResource {
     const { status, externalVerificationRedirectURL } = firstFactorVerification;
 
     if (status === 'unverified' && externalVerificationRedirectURL) {
-      windowNavigate(externalVerificationRedirectURL);
+      navigateCallback(externalVerificationRedirectURL);
     } else {
       clerkInvalidFAPIResponse(status, SignIn.fapiClient.buildEmailAddress('support'));
     }
   };
 
+  public authenticateWithRedirect = async (params: AuthenticateWithRedirectParams): Promise<void> => {
+    return this.authenticateWithRedirectOrPopup(params, windowNavigate);
+  };
+
+  public authenticateWithPopup = async (params: AuthenticateWithPopupParams): Promise<void> => {
+    const { popup } = params || {};
+    if (!popup) {
+      clerkMissingOptionError('popup');
+    }
+
+    return this.authenticateWithRedirectOrPopup(params, url => {
+      popup.location.href = url instanceof URL ? url.toString() : url;
```

Comment:
[nit] if the `url` param can be a URL or a string, then we can just safely call `url.toString()` in both cases and remove the conditional logic